lint-rules.md

```diff
@@ -0,0 +1,1280 @@
+
+### `blanks_between_elements`
+
+There should be a blank line between each WDL element at the root indentation level (such as the import block and any task/workflow definitions) and between sections of a WDL task or workflow. Never have a blank line when indentation levels are changing (such as between the opening of a workflow definition and the meta section). There should also never be blanks _within_ a meta, parameter meta, input, output, or runtime section. See example for a complete WDL document with proper spacing between elements. Note the blank lines between meta, parameter meta, input, the first call or first private declaration, output, and runtime for the example task. The blank line between the workflow definition and the task definition is also important.
+
+**Group**: `spacing`
+
+#### Example
+
+Good:
+
+```wdl
+version 1.1
+
+import "../../tools/unused-example.wdl" as unused_example
+
+workflow hello_world {
+    meta {
+        description: "Greets the user of the workflow."
+        output: {
+            statement: "The WDL generated statement for the user.",
+        }
+    }
+
+    parameter_meta {
+        name: "Name to greet"
+        greeting: "Phrase to use while greeting `name`"
+    }
+
+    input {
+        String name = "World"
+        String greeting = "Hello"
+    }
+
+    call greet { input:
+        name = name,
+        greeting = greeting,
+    }
+
+    output {
+        String statement = greet.statement
+    }
+}
+
+task greet {
+    meta {
+        description: "Generates a statement from a name and a greeting"
+        output: {
+            statement: "The result of combining the input name and greeting",
+        }
+    }
+
+    parameter_meta {
+        name: "Name to greet"
+        greeting: "Phrase to use while greeting `name`"
+    }
+
+    input {
+        String name
+        String greeting
+    }
+
+    String statement = "~{greeting}, ~{name}"
+
+    command <<<
+        echo "~{statement}" > statement.txt
+    >>>
+
+    output {
+        String statement = read_string("statement.txt")
+    }
+
+    runtime {
+        memory: "4 GB"
+        disks: "10 GB"
+        container: "docker://ghcr.io/stjudecloud/util@sha256:c0583fe91d3e71fcfba58e2a57beb3420c7e907efd601f672fb5968086cd9acb"  # tag: 1.3.0
+        maxRetries: 1
+    }
+}
+
+```
+
+### `inconsistent_newlines`
+
+Files should not mix `\n` and `\r\n` line breaks. Pick one and use it consistently in your project.
+
+### `line_width`
+
+WDL lines should be less than or equal to 90 characters wide whenever possible. This line width restriction applies to embedded code in the `command` block. Exceptions would be long strings that WDL doesn't allow to be broken up within the meta and parameter meta sections. Another exception is `container` lines inside the `runtime` block of a task. (See the rules `mutable_container`  && `immutable_container_not_tagged` for more information about permitted `container` lines.)
+
+### `expression_spacing`
```

Comment:
- There should be a compact vs more newlines way to toggle the behavior of this lint to enforce "only one way to do something" for repos to automate... Maybe two separate lints that are incompatible?